functions/go/fix/fixpkg/fix.go

```diff
@@ -0,0 +1,453 @@
+
+	// convert packageMetadata in v1alpha1 Kptfile to v1alpha2 info
+	if kfOld.PackageMeta != nil {
+		emails := []string{kfOld.PackageMeta.Email}
+		if kfOld.PackageMeta.Email == "" {
+			emails = nil
+		}
+		kfNew.Info = &v1alpha2.PackageInfo{
+			Site:        kfOld.PackageMeta.URL,
+			Emails:      emails,
+			License:     kfOld.PackageMeta.License,
+			Description: kfOld.PackageMeta.ShortDescription,
+			Keywords:    kfOld.PackageMeta.Tags,
+			Man:         kfOld.PackageMeta.Man,
+		}
+	}
+
+	// convert upstream section
+	if kfOld.Upstream != nil {
+		kfNew.Upstream = &v1alpha2.Upstream{
+			Type: v1alpha2.OriginType(kfOld.Upstream.Type),
+			Git: &v1alpha2.Git{
+				Repo:      kfOld.Upstream.Git.Repo,
+				Directory: kfOld.Upstream.Git.Directory,
+				Ref:       kfOld.Upstream.Git.Ref,
+			},
+			UpdateStrategy: v1alpha2.ResourceMerge,
+		}
+
+		kfNew.UpstreamLock = &v1alpha2.UpstreamLock{
+			Type: v1alpha2.OriginType(kfOld.Upstream.Type),
+			Git: &v1alpha2.GitLock{
+				Repo:      kfOld.Upstream.Git.Repo,
+				Directory: kfOld.Upstream.Git.Directory,
+				Ref:       kfOld.Upstream.Git.Ref,
+			},
+		}
+	}
+
+	// convert OpenAPI section in v1alpha1 Kptfile to apply-setters
+	s.settersSchema, err = schemaUsingField(node, openapi.SupplementaryOpenAPIFieldName)
+	if err != nil {
+		return node, err
+	}
+	setters, err := listSetters(node)
+	if err != nil {
+		return node, err
+	}
+
+	pl := &v1alpha2.Pipeline{}
+	pl.Mutators = append(pl.Mutators, functions...)
+	kfNew.Pipeline = pl
+
+	if setters != nil && len(setters) > 0 {
+		fn := v1alpha2.Function{
+			Image:     "gcr.io/kpt-fn/apply-setters:v0.1",
+			ConfigMap: setters,
+		}
+		pl.Mutators = append(pl.Mutators, fn)
+	}
+
+	// convert inventory section
+	if kfOld.Inventory != nil {
+		kfNew.Inventory = &v1alpha2.Inventory{
+			Namespace:   kfOld.Inventory.Namespace,
+			Name:        kfOld.Inventory.Name,
+			InventoryID: kfOld.Inventory.InventoryID,
+			Labels:      kfOld.Inventory.Labels,
+			Annotations: kfOld.Inventory.Annotations,
+		}
+	}
+
+	// convert kfNew to yaml node
+	b, err := yaml.Marshal(kfNew)
+	if err != nil {
+		return node, err
+	}
+	kNode, err := yaml.Parse(string(b))
+	if err != nil {
+		return node, err
+	}
+	err = kNode.SetAnnotations(meta.Annotations)
+	return kNode, err
+}
+
+// FunctionsInPkg gets the v1alpha2 functions list for functions in package
+// nodes is list of input nodes which are sorted according to the package depth
+// i is the index of the Kptfile of the package, all the files till i hits next Kptfile
+// are the files of the package
+// pkgPath is the package path relative to the root package directory
+func FunctionsInPkg(nodes []*yaml.RNode, i int, pkgPath string) []v1alpha2.Function {
+	var res []v1alpha2.Function
+	for i < len(nodes) {
+		node := nodes[i]
+		meta, err := node.GetMeta()
+		if err != nil {
+			return res
+		}
+		if meta.Kind == v1alpha2.KptFileName {
+			// we hit the Kptfile of next package, so break here
+			break
```

Comment:
Do we know that the Kptfile will always be the first file encountered for every package?

Reply:
Refer to previous comment.